state/machine.go

```diff
@@ -2069,7 +2070,7 @@ func (m *Machine) CreateUpgradeSeriesPrepareLock() error {
 		}
 		locked, err := m.IsLocked()
 		if err != nil {
-			return nil, errors.Wrap(err, jujutxn.ErrNoOperations)
```

Comment:
Did we add code to machine.Destroy and the cleanup code to ensure that these documents get deleted? I don't think we want to prevent 'juju remove-machine' from tearing down the machine, because we decided that would be our "something went wrong" button, but we should make sure we don't end up with garbage docs in database.

Reply:
Yes, that is so important in fact that an entire PR/kanban card will be dedicated to getting that correct. If the machine is removed for this doc will be deleted as well.

Reply:
great